### ISSUE TITLE
[Bug] Multiple Interfaces can break Objects-Plugin

### ISSUE THREAD

**Issue by j-riebe** [jhnnsrs/turms#24]
##  Problem
I tried the objects-plugin and I must say once again **nice work**!

Unfortunately, there is a problem when modeling complex inheritance and composition.

The codegen works as expected, but the generated code can't be executed due to problems with pythons MRO.

As a workaround, one can write the SDL in the correct order, because the generators keeps the order of the interface definition in the SDL.

## Minimal example:
```gql
interface A { 
  foo: String!
}

interface B implements A { 
  foo: String!
  bar: String!
 }

type ThisWork implements B & A { 
  foo: String!
  bar: String!
}

type ThisDoesnt implements A & B { 
  foo: String!
  bar: String!
}
```
translates to
```python
class A:
    foo: str

class B(A):
    foo: str
    bar: str

class ThisWorks(B, A):
    foo: str
    bar: str

class ThisDoesnt(A, B):  # TypeError: Cannot create a consistent method resolution order (MRO) for bases A, B
    foo: str
    bar: str

```

## Possible solution

Basically, the generator must be changed to either:
a) Sort the Superclasses according to their inheritance-structure (rather complex): ```class ThisWorks(B, A)``` 
b) only generate the final classes of the inheritance/composition chain: ```class ThisWorksToo(B)```

An entrypont to b):
```python 
class A:
    ...

class B(A):
    ...

class C(B):
    ...

class D(A):
    ...

# type E implements A & B & C & D

base_classes = {A, B, C, D}
base_classes_to_generate = base_classes.copy()
for cls in base_classes:
    other_classes = base_classes - {cls}
    if any([issubclass(possible_subclass, cls) for possible_subclass in other_classes]):
        base_classes_to_generate.remove(cls)

>>> base_classes_to_generate == {C, D}
```

**Comment by j-riebe** [comment]
And I just found a second issue, that is related to interfaces. (So I'll just leave it here 😉  )

```python
class DataSourceBase(BaseModel):
    forward: "Reference"

class Reference(BaseModel):
    ...

DataSource = Union[Type1, Type2]
DataSource.update_forward_refs()  # This should be DataSourceBase.update_forward_refs()
```
